src/agents/k8s/tools/query.py

```
@@ -30,13 +31,35 @@ async def k8s_query_tool(
     The URI must follow the format of Kubernetes API.
     The returned data is sanitized to remove any sensitive information.
     For example, it will always remove the `data` field of a `Secret` object."""
-    result = await k8s_client.execute_get_api_request(uri)
-    if not isinstance(result, list) and not isinstance(result, dict):
-        raise Exception(
-            f"failed executing k8s_query_tool with URI: {uri}."
-            f"The result is not a list or dict, but a {type(result)}"
-        )
-    return result
+    try:
+        result = await k8s_client.execute_get_api_request(uri)
+        if not isinstance(result, (list, dict)):
+            raise K8sClientError(
+                message=f"Invalid result type: {type(result)}",
+                status_code=500,
+                uri=uri,
+                tool_name="k8s_query_tool",
+            )
+        return result
+    except K8sClientError as e:
```

this `K8sClientError` is defined by us, we know where it gets raised in the codebase, so it's better to assign the `tool_name` in each place when this error gets raised, instead of mutating the error in the exception logic and raising it again.

---

The K8sClientError with empty tool_name from the client layer is intentional, it allows the client to be reusable across different tools, and each tool explicitly sets its own tool_name when handling the error. This provides better separation of concerns.